### PR TITLE
Adding netcore50 TFM

### PR DESCRIPTION
**Pull request by pranavkm**
cc @natemcmaster

### CODE REVIEW

File: src/EntityFramework.MicrosoftSqlServer/project.json [hunk 1]
```diff
@@ -25,6 +25,12 @@
         "System.Threading.Thread": "4.0.0-*",
         "System.Threading.ThreadPool": "4.0.10-*"
       }
+    },
+    "netcore50": {
+      "dependencies": {
+        "System.Data.SqlClient": "4.0.0-*",
+        "System.Threading.Thread": "4.0.0-*"
```
Comment (natemcmaster) [inline src/EntityFramework.MicrosoftSqlServer/project.json]: Don't we need to use the RTM package versions?

Comment (pranavkm) [inline src/EntityFramework.MicrosoftSqlServer/project.json]: I essentially undid the changes from my previous commit: https://github.com/aspnet/EntityFramework/commit/7e0ba479a2cc1242ff7d6688687a740bd8a606f4#diff-0346fce18fd5fb2d9f6971bfa089439aL28

Comment (natemcmaster) [inline src/EntityFramework.MicrosoftSqlServer/project.json]: Ok. We may need to change this in the future, but SqlClient doesn't work on UWP anyways right now.

**Comment by natemcmaster** [comment]
:shipit:

**Comment by natemcmaster** [comment]
Fixes #3517

**Comment by natemcmaster** [comment]
@pranavkm have all of EF's dependencies now got netcore50? If so, this should build w/o trouble.

**Comment by pranavkm** [comment]
Checked in as part of https://github.com/aspnet/EntityFramework/commit/78559e6321aec1d6b0aa67abb08673452235d99e

**Pull request closed without merging**